### ISSUE TITLE
bc_email_on_started not working

### ISSUE THREAD

**Issue by btsherid** [OSC/ondemand#760]
Hi,

When using the bc_email_on_started check box I never receive an email. I have tested that email is working by doing the following:

Added to form.yml:
Under attributes:
  email:
    label: "Email address"
    help: "Enter email address if you want to receive an email when job starts"
Under form:
  - email
  - email_on_started

Added to submit.yml.erb under script:
  email_on_started: <%= ! email_on_started.to_i.zero? %>
  <% unless email.blank? %>
  email: <%= email %>
  email_on_started: true
  <% end %>

This gives me a text box where I can type in my email address. This successfully sends me an email for both started and terminated (if I change the above from started to terminated).

Any idea where I can start debugging/troubleshooting why this does not work with the bc_email_on_started  check box? I'm assuming it has something to with the value that gets set for the users email when they login with LDAP but I don't know where/how to find that value.

Thanks,
Brendan Sheridan

**Comment by johrstrom** [comment]
Sorry I keep transferring this here and there.  This seems to be the right repo because you're having issue with the dashboards `bc_` flags.

But what I'm seeing there is it's working when you set the `email` attribute. Maybe it's something strange with the schedulers default choice for your email.  That is, have you tried `bc_email_on_started` _with_ `email` as well?  What type of scheduler are you using?

Is this the right understanding of the combination of attributes and if they work?
`bc_email_on_started`  - doesn't work
`bc_email_on_started` & `email` - does this work?
`email_on_started` & `email` - works

**Comment by btsherid** [comment]
bc_email_on_started - doesn't work
bc_email_on_started & email - works
email_on_started & email - works

This is the setup I used for bc_email_on_started:
<pre>
Added to form.yml:
Under attributes:
email:
label: "Email address"
help: "Enter email address if you want to receive an email when job starts"
Under form:

- email
- email_on_started
Added to submit.yml.erb under script:
bc_email_on_started: <%= ! bc_email_on_started.to_i.zero? %>
<% unless email.blank? %>
email: <%= email %>
bc_email_on_started: true
<% end %>
</pre>

Thanks,
Brendan

**Comment by johrstrom** [comment]
OK. What's the scheduler you use? I'm sure this is something on it's side.

Here's what would happen for Slurm:
`bc_email_on_started` & `email` would generate `--mail-type BEGIN --mail-user johrstrom@osc.edu` arguments to `sbatch`.

Without email it would only generate  `--mail-type BEGIN` command arguments and let the Slurm controller determine what my email address is (likely based of of my user & ldap records, though I'm not 100% sure).

I'd say look into replicating the same from the command line. submit a very simple script to your scheduler and determine if it can correctly email you when the job's started when you don't give it the email address argument (whatever that may be for your scheduler).

**Comment by btsherid** [comment]
Thanks. We are using SLURM. I submitted a simple job using srun with --mail-type BEGIN and I didn't get an email. I added in --mail-user and I did get an email. I can't find any documentation for SLURM to automatically determine email addresses. Everything I've found says to set --mail-user in the job script.

Is there any option in OnDemand to run a bash command and use the output in the form.yml? What I'm thinking of is running a command on the OnDemand server that will determine the user's email address and save that to an environment variable. Then set email: "USERNAME_VAR" in form.yml

If not, I honestly don't think this feature will be used that much and I'll probably just disable it.

Thanks,
Brendan
